### PR TITLE
[Eager] inc ref before return Py_None

### PR DESCRIPTION
**Pull request by wanghuancoder**
### PR types

Bug fixes 
### PR changes

Others 
### Describe

在return Py_None前，必须先Py_INCREF(Py_None);

### CODE REVIEW

File: paddle/fluid/pybind/eager_method.cc [hunk 1]
```diff
@@ -1218,6 +1218,7 @@ static PyObject* tensor__set_grad_type(TensorObject* self, PyObject* args,
   } else if (var_type == framework::proto::VarType::SELECTED_ROWS) {
     grad_tensor->set_impl(std::make_shared<phi::SelectedRows>());
   }
+  Py_INCREF(Py_None);
```
Comment (JiabinYang) [inline paddle/fluid/pybind/eager_method.cc]: Maybe it's good to seal an API for others

**Review by JiabinYang — DISMISSED** [review]
LGTM

**Review by JiabinYang — APPROVED** [review]
LGTM

**Review by zhiqiu — APPROVED** [review]
LGTM

**Pull request merged**